src/Tuple.js

```diff
@@ -0,0 +1,74 @@
+function Tuple(x, y) {
+    switch (arguments.length) {
+        case 0:
+            throw new TypeError('no arguments to Tuple');
+        case 1:
+            return function(y) {
+                return new _Tuple(x, y)
+            };
+        default:
+            return new _Tuple(x,y)
+    }
+};
+
+function _Tuple(x, y) {
+    this[0] = x;
+    this[1] = y;
+    this.length = 2;
+};
+
+function ensureConcat() {
+    var xs = [].slice.call(arguments)
```

Comment:
since this is a privateh helper anyways, maybe we should just pass a list, rather than arg-slicing